Update Python Curriculum Page

annamtong opened the pull request:
Updated design of the Python Curriculum Page (modified the html and sass pages) and cleaned it up based upon the mockup from #198. After talking to Einar, we decided to leave some aspects of the mockup design out though. The modifications included adding an image to the page, updating the colors and spacing of some elements to make the page look cleaner, as well as splitting the lesson cards up by group/category.  

Closes #198.

mattxwang commented on classes/py1.html:
```diff
@@ -62,6 +65,43 @@ <h2><a class='py1-link' href="{{ lesson.url }}">{{ lesson.title }}</a></h2>
+            {% endfor %}
+        </div>
+
+        <!-- flow lesson cards -->
+        <h2 class = "title">flow</h2>
+        <div class='py1-lessons-wrapper'>
+            {% for lesson in site.py1 %}
+            {% if lesson.group == "flow" %}
+            <div class='py1-lesson'>
+                <div class='py1-lesson-title-bar'>
+                    <h2><a class='py1-lesson-link' href="{{ lesson.url }}">{{ lesson.title }}</a></h2>
+                    <p class='py1-step-number'>#{{ lesson.num }}</p>
+                </div>
+                <div class='py1-lesson-body'>
+                    <ul class='py1-list py1-lesson-goal-list'>
+                        {% for brief in lesson.agenda %}
+                        <li class='py1-lesson-goal'><span>{{ brief.text }}</span></li>
+                        {% endfor %}
+                    </ul>
+                </div>
+                <div class='py1-lesson-action-bar'>
+                    <div class='py1-action-group'>
+                        {% if lesson.slides_link %}
+                        <a class='py1-lesson-action' href="{{ lesson.slides_link }}">slides</a>
+                        {% endif %}
+                        {% if lesson.video_link %}
+                        <a class='py1-lesson-action' href="{{ lesson.video_link }}">video</a>
+                        {% endif %}
+                        {% if lesson.worksheet_link %}
+                        <a class='py1-lesson-action' href="{{ lesson.worksheet_link }}">worksheet</a>
+                        {% endif %}
+                    </div>
+                    <p class="py1-lesson-duration">{{ lesson.length }} min</p>
+                </div>
+            </div>
+            {% endif %}
```
This is quite a bit of duplicated code with the previous section! Unfortunately, we can't really reformat the way the data is stored (as it is a collection), but we could instead create some sort of reusable component to represent this card. 

The Jekyll-based way we could do this is with an `_includes` "component", which we can then pass in information to render the card. I believe Jun has created something similar in [`py1-lesson-card-content`](https://github.com/uclaacm/teach-la-website/blob/master/_includes/py1-lesson-card-content.html), which you can check out! If you want an example of an include designed around parameters, you can check out [`blog-author-card.html`](https://github.com/uclaacm/teach-la-website/blob/master/_includes/blog-author-card.html); [here are the relevant docs on includes](https://jekyllrb.com/docs/includes/).

annamtong replied:
Thanks so much! I was having trouble figuring out how to consolidate the duplicated code and just tried creating an `_includes` that consists of all the duplicated code and a parameter specifying the lesson group.

mattxwang commented on classes/py1.html:
```diff
@@ -6,9 +6,9 @@
-        <span class="py1-link">python</span>
-        </h1>
+        <span class = "py1-smaller-margin-p">how to get started in</span>
+        <span class= "py1-smaller-margin-heading"><span class = "py1-link">python</span></span>
```
A minor piece of feedback, but I think it's a bit confusing to use our link styling if this thing isn't actually a link. If we want the underline effect (without the hover), we can implement that independently; however, it might also be worth picking a style that doesn't clash with our link styling.

annamtong replied:
I tried a few different options and ended up deciding to leave out any effects here. With the added image directly below the title, I thought that it looked cleaner without anything, but am open to suggestions if you or @EinarBalan think including some sort of effect would look nicer!

mattxwang commented on classes/py1.html:
```diff
@@ -33,11 +33,14 @@ <h1>how to get started in<br />
             <div class='py1-lesson'>
                 <div class='py1-lesson-title-bar'>
-                    <h2><a class='py1-link' href="{{ lesson.url }}">{{ lesson.title }}</a></h2>
+                    <h2><a class='py1-lesson-link' href="{{ lesson.url }}">{{ lesson.title }}</a></h2>
```
This should probably be an `<h3>`, since it is under the `<h2>` category "fundamentals". Similar comment on the flow ones!

mattxwang commented on _sass/_py1.scss:
```diff
@@ -96,6 +96,15 @@
     padding-left: 0.5em;
 }
 
+.py1-smaller-margin-p {
```
I believe we have a `.no-margin` class that already does this!

mattxwang commented on classes/py1.html:
```diff
@@ -6,9 +6,9 @@
-        <h1>how to get started in<br />
-        <span class="py1-link">python</span>
-        </h1>
+        <span class = "py1-smaller-margin-p">how to get started in</span>
```
Is there a reason you removed this from being an `<h1>`? It doesn't seem like there's one on the page anymore!

annamtong replied:
Was attempting to resolve a margins issue, but fixed it through using the `.no-margin` class and added the `<h1>` back!

mattxwang reviewed (changes requested):
Hi Anna,

Thanks for all your work on this page, I love what you've done with it so far, and thank you for working closely with Einar! 

I've left a few comments, mostly around semantic HTML and code duplication; let me know if you need any clarification!

Also, one thing that doesn't really make it into a comment: currently if a lesson has a slide, video, and worksheet, the card gets too cramped and the time estimate looks a bit strange. I think it's up to you on how you want to resolve this (ex keep the same, move time to a new row, make the indicator links smaller), but do what makes the most sense to you!

annamtong commented:
Made changes to the Python Curriculum Page addressing the issues of duplicated code and semantic HTML. Created a new [page](_includes/py1-lesson-card.html) under the `_includes` folder to address code duplication, fixed issues with some of the headings, and removed unnecessary link styling.

mattxwang reviewed (approved):
Thanks for the quick turnaround Anna! The way you did the `include` is pretty elegant, major respect 😊 

On my end, I think this is good to merge. Let's wait for one last approving review from @EinarBalan, and I think it's good to go! And make sure to use the squash and merge option. Great work Anna!

annamtong commented:
Thanks so much for all your suggestions and help @mattxwang !

EinarBalan reviewed (approved):
Lgtm!
